File: app/src/main/java/quintype/com/templatecollectionwithrx/ui/main/viewholders/SectionChildViewHolder.java
package quintype.com.templatecollectionwithrx.ui.main.viewholders;

import android.view.View;
import android.widget.TextView;

import com.bignerdranch.expandablerecyclerview.ViewHolder.ChildViewHolder;

import quintype.com.templatecollectionwithrx.R;
import quintype.com.templatecollectionwithrx.models.NavMenu;

public class SectionChildViewHolder extends ChildViewHolder {

    public TextView mSubsectionTextView;

    /**
     * Default constructor.
     *
     * @param itemView The {@link View} being hosted in this ViewHolder
     */
    public SectionChildViewHolder(View itemView) {
        super(itemView);
        mSubsectionTextView = (TextView) itemView.findViewById(R.id.child_section_name);
    }

    public void bind(NavMenu subSection) {
        mSubsectionTextView.setText(subSection.title());
    }
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/ui/main/viewholders/SectionChildViewHolder.kt
package quintype.com.templatecollectionwithrx.ui.main.viewholders

import android.view.View
import android.widget.TextView
import com.bignerdranch.expandablerecyclerview.ViewHolder.ChildViewHolder
import quintype.com.templatecollectionwithrx.R
import quintype.com.templatecollectionwithrx.adapters.DrawerSectionsAdapter
import com.example.androidcore.models.NavMenu
import quintype.com.templatecollectionwithrx.ui.main.navmenu.NavMenuGroup

class SectionChildViewHolder(itemView: View?) : ChildViewHolder(itemView) {

    public var mSubsectionTextView: TextView? = null
    var parent: NavMenuGroup? = null
    var mListener: DrawerSectionsAdapter.OnDrawerItemSelectedListener? = null

    constructor(itemView: View?, listener: DrawerSectionsAdapter.OnDrawerItemSelectedListener?) : this(itemView) {
        mSubsectionTextView = itemView?.findViewById<View>(R.id.child_section_name) as TextView
        mListener = listener
    }

    fun bind(subSection: NavMenu) {
        mSubsectionTextView?.text = subSection.title()
    }
}